app/src/main/java/com/teamltt/carcare/activity/ReminderEditActivity.java

```diff
@@ -17,52 +17,195 @@
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        spinner.setAdapter(adapter);
+
+        if (reminderId >= 0) {
+            Reminder reminder = dbHelper.getReminderByReminderId(reminderId);
+            ((TextView) findViewById(R.id.field_reminder_name)).setText(reminder.getName());
+            //check feature id to see if reminder checks for date or for a feature
+            if (reminder.getFeatureId() >= 0) { //if checks for feature, set feature type spinner, value, and comparison type spinner
+                ((Spinner) findViewById(R.id.spinner_feature)).setSelection(0); //TODO: add a way to automatically switch to correct spinner choice from feature id
+                ((Spinner) findViewById(R.id.spinner_feature)).setSelection(reminder.getComparisonType());
+                ((EditText) findViewById(R.id.field_value)).setText(""+reminder.getComparisonValue());
+            } else {
+                ((Spinner) findViewById(R.id.spinner_reminder_type)).setSelection(1); //if checks for date, set the spinner to date, populate with date
+                ((TextView) findViewById(R.id.text_date)).setText(reminder.getDate());
+            }
+
+        }
+    }
+
+    private boolean saveReminder() {
+        String name = ((EditText) findViewById(R.id.field_reminder_name)).getText().toString();
+        int comparison = ((Spinner) findViewById(R.id.spinner_comparison)).getSelectedItemPosition();
+        String valueStr = ((EditText) findViewById(R.id.field_value)).getText().toString();
+        int value;
+        if (valueStr.equals("")) {
+            value = -1;
+        } else {
+            value = Integer.parseInt(valueStr);
+        }
+
+        if (reminderId < 0) {
+            Log.i(TAG, "adding new reminder");
+            long status = dbHelper.createNewReminder(vehicleId, name, featureId, comparison, value, formattedDate);
+            return status > 0;
+        } else {
+            Log.i(TAG, "editing reminder of id " + reminderId);
+            long status = dbHelper.updateReminder(reminderId, vehicleId, name, featureId, comparison, value, formattedDate);
+            return status > 0;
+        }
+    }
+
+    public void showDatePickerDialog(View view) {
+        DatePickerFragment dialog = new DatePickerFragment();
+        dialog.show(getFragmentManager(), "DateFragment");
+    }
+
+    public void setDate(int year, int month, int day) {
+        TextView text = (TextView) findViewById(R.id.text_date);
+        String date = month + "/" + day + "/" + year;
+        text.setText(date);
+        String formattedMonth = "" + month;
+        String formattedDay = "" + day;
+        if (formattedMonth.length() < 2) {
+            formattedMonth = "0" + formattedMonth;
+        }
+        if (formattedDay.length() < 2) {
+            formattedDay = "0" + formattedDay;
+        }
+        formattedDate = year + "-" + formattedMonth + "-" + formattedDay + " 00:00:00";
```

Comment:
It's what eventually gets put in the Reminder after creation. The rest of a Reminder's values are taken straight from the EditTexts but since the date is displayed for the user as MM/dd/yyyy and the database takes dates as yyyy-MM-dd HH:mm:ss I save the formatted date here and use it later in saveReminder